### ISSUE TITLE
[Suggestion] Add metadata specifier to UPROPERTY/UCLASS/USTRUCT etc.

### ISSUE THREAD

**Issue by kadium91** [UE4SS-RE/RE-UE4SS#561]
Hello, would it be possible to add metadata specifiers to dumped properties/classes/structs ? (I'm talking about the meta stuff like meta(ClampMin=..., ClampMax=...), you can find a sample on FColor.), also maybe add Category property specifier (if available, this might be exposed only to editor builds idk)

You have a sample of how to get metadata in "UField::GetMetaData" in Engine/Source/Runtime/CoreUObject/Private/UObject/Class.cpp

**Comment by UE4SS** [comment]
Unfortunately the MetaData related functions and variables only exist in the UE editor, so the only way to do this would be to glean which specifiers are appropriate from the context of the property at runtime which can be difficult and inaccurate.
I appreciate letting us know where to look though, it's very helpful.

**Comment by kadium91** [comment]
> Unfortunately the MetaData related functions and variables only exist in the UE editor, so the only way to do this would be to glean which specifiers are appropriate from the context of the property at runtime which can be difficult and inaccurate. I appreciate letting us know where to look though, it's very helpful.

You're right, didn't saw that was marked EDITORONLY. There might be a way tho as I know some of metadata are used at runtime in production build.

**Comment by kadium91** [comment]
> Unfortunately the MetaData related functions and variables only exist in the UE editor, so the only way to do this would be to glean which specifiers are appropriate from the context of the property at runtime which can be difficult and inaccurate. I appreciate letting us know where to look though, it's very helpful.

After talking with a former epic employee here is what he told me about accessing metadata in shipping builds

> Not without engine changes.  Some metadata makes it into property flags but most doesn’t
Package level metadata can be promoted to asset registry tags without modifying the engine though

The specific metadata specifier I was interested to was AssetBundles btw. I think we could still get it apart from metadata since they are used at runtime for shipping builds. I'll look at it tmr.

**Comment by narknon** [comment]
It says it's NULL in the game.
